awareness.js

```diff
@@ -189,9 +237,10 @@ export const removeAwarenessStates = (awareness, clients, origin) => {
 /**
  * @param {Awareness} awareness
  * @param {Array<number>} clients
+ * @param {DefaultAwarenessStateEncoder|BinaryAwarenessStateEncoder} stateEncoder The encoder to use for encoding and decoding each state entry
  * @return {Uint8Array}
  */
-export const encodeAwarenessUpdate = (awareness, clients, states = awareness.states) => {
+export const encodeAwarenessUpdate = (awareness, clients, states = awareness.states, stateEncoder = new DefaultAwarenessStateEncoder()) => {
```

Comment:
It should not be necessary to create a new instance of anything here. `*AwarenessEncoder` should only have static methods.